routing/routing_integration_tests/concurrent_feature_parsing_test.cpp

```diff
@@ -0,0 +1,104 @@
+#include "testing/testing.hpp"
+
+#include "routing/routing_integration_tests/routing_test_tools.hpp"
+
+#include "indexer/classificator_loader.hpp"
+#include "indexer/data_source.hpp"
+
+#include "platform/country_file.hpp"
+#include "platform/local_country_file.hpp"
+
+#include "geometry/point2d.hpp"
+
+#include "base/logging.hpp"
+
+#include <algorithm>
+#include <functional>
+#include <future>
+#include <mutex>
+#include <string>
+#include <thread>
+#include <vector>
+
+namespace
+{
+using namespace platform;
+using namespace std;
+
+void TestConcurrentAccessToFeatures(string const & mwm)
+{
+  FrozenDataSource dataSource;
+
+  auto const & countryFile = CountryFile(mwm);
+  auto const & local = integration::GetLocalCountryFileByCountryId(countryFile);
+  TEST_NOT_EQUAL(local, LocalCountryFile(), ());
+  TEST(local.HasFiles(), (local));
+
+  auto const mwmIdAndRegResult = dataSource.RegisterMap(local);
+  TEST_EQUAL(mwmIdAndRegResult.second, MwmSet::RegResult::Success, (local));
+  TEST(mwmIdAndRegResult.first.IsAlive(), (local));
+
+  auto const handle = dataSource.GetMwmHandleByCountryFile(countryFile);
+  TEST(handle.IsAlive(), (local));
+
+  auto const featureNumber = handle.GetValue()->m_table->size();
+  // Note. If hardware_concurrency() returns 0 it means that number of core is not defined.
+  // If hardware_concurrency() returns 1 it means that only one core is available.
+  // In the both cases 2 threads should be used for this test.
```

Comment:
maybe hardcode it to 2 and get rid of this explanation?

Reply:
It's not only example but also a test. If we change something in our codebase this test may show us it. On my MacBook hardware_concurrency() returns 12 (6 core and hyperthreading). I guess the more real threads the more chances to find a wrong behavior. So I'm for leaving this code as is. What do you think?

As for the comment it's just for underline the behavior of `thread::hardware_concurrency()`. I noticed sometimes developers don't know that it may return 0.